execution_engine/src/core/engine_state/upgrade.rs

```diff
@@ -281,7 +257,8 @@ where
             .check_next_version(&self.new_protocol_version)
             .is_major_version();
 
-        if !self.has_different_entry_points(&contract, &entry_points) && !is_major_bump {
+        let entry_points_unchanged = *contract.entry_points() == entry_points;
```

Comment:
Oh, I see you removed the `has_different_entry_points` entirely and make use of `impl Eq for EntryPoints` - I don't know what's the performance of that but hopefuly Rust also produces some smart code for the equality.